### PR TITLE
Tabular: Removed usage of catboost-dev

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Removed usage of catboost-dev (can just use catboost for soft-class now that 0.24 is released)
- NOTE: softclass is extremely slow for catboost due to unoptimized objective and metric for soft log loss. The below warnings are printed when trying to train with softclass (although it does train successfully, just 10x slower than would be expected).

```
Fitting model: CatBoost_DSTL ...
	Fitting CatBoost_DSTL with 'num_gpus': 0, 'num_cpus': 4
	Catboost model hyperparameters: {'iterations': 10000, 'learning_rate': 0.05, 'random_seed': 0, 'allow_writing_files': False, 'loss_function': <autogluon.tabular.models.catboost.catboost_softclass_utils.SoftclassObjective.SoftLogLossObjective object at 0x103bceb00>, 'eval_metric': <autogluon.tabular.models.catboost.catboost_softclass_utils.SoftclassCustomMetric.SoftLogLossMetric object at 0x11ba446a0>}
/Users/neerick/workspace/venvs/autogluon/lib/python3.6/site-packages/catboost/core.py:1464: UserWarning: Failed to optimize method "calc_ders_multi" in the passed object:
Failed in nopython mode pipeline (step: nopython frontend)
Untyped global name 'sum': Cannot determine Numba type of <class 'builtin_function_or_method'>
File "tabular/src/autogluon/tabular/models/catboost/catboost_softclass_utils.py", line 50:
        def calc_ders_multi(self, approxes, targets, weight):
            <source elided>
            exp_approx = [math.exp(val) for val in approxes]
            exp_sum = sum(exp_approx)
            ^
  self._object._train(train_pool, test_pool, params, allow_clear_pool, init_model._object if init_model else None)
/Users/neerick/workspace/venvs/autogluon/lib/python3.6/site-packages/catboost/core.py:1464: UserWarning: Failed to optimize method "evaluate" in the passed object:
Failed in nopython mode pipeline (step: nopython frontend)
Untyped global name 'soft_log_loss': Cannot determine Numba type of <class 'autogluon.core.metrics._ProbaScorer'>
File "tabular/src/autogluon/tabular/models/catboost/catboost_softclass_utils.py", line 36:
        def evaluate(self, approxes, target, weight):
            <source elided>
            approxes = np.multiply(approxes, 1/np.sum(approxes, axis=1)[:, np.newaxis])
            error_sum = soft_log_loss(np.array(target), np.array(approxes))
            ^
  self._object._train(train_pool, test_pool, params, allow_clear_pool, init_model._object if init_model else None)
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jwmueller** [comment]
Pushed a JIT compatible soft logloss to avoid these warnings

**Comment by szha** [comment]
Job PR-1338-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1338/2/index.html
